docs/dev/architeture.md

```diff
@@ -0,0 +1,206 @@
+
+#### SST
+module path: https://github.com/CeresDB/ceresdb/tree/main/analytic_engine/src/sst
+
+Both `Flush` and `Compaction` involves `SST` and in the codebase `SST` itself is actually an abstraction that can have multiple specific implementations. The current implementation is based on [Parquet](https://parquet.apache.org/), which is a column-oriented data file format designed for efficient data storage and retrieval.
+
+The format of `SST` is very critical for retrieving data and is also the most important part to perform well in handling both timeseries and analytic workloads. At present, our [Parquet](https://parquet.apache.org/)-based implementation is good at processing analytic workload but is poor at processing timeseries workload. In our road map, a new hybrid storage format is planned to be able to process the two workloads at the same time.
+
+#### Space
+module path: https://github.com/CeresDB/ceresdb/blob/main/analytic_engine/src/space.rs
+
+In `Analytic Engine`, there is a concept called `space` and here is an explanation for it to resolve some ambiguities when read source code. Actually `Analytic Engine` does not have the concept of `catalog` and `schema` and only provides two levels of relationship: `space` and `table`. And in the implementation, the `schema id` (which should be unique across all `catalog`s) on the upper layer is actually mapped to `space id`.
+
+The `space` in `Analytic Engine` serves mainly for isolation of resources for different tenants, such as the usage of memory.
+
+## Critical Path
+After a brief introduction to some important modules of CeresDB, we will give a description for some critical paths in code, hoping to provide interested developers with a guide for reading the code.
+
+### Query
+Take `SELECT` SQL as an example:
+- Server module chooses a proper module to process the requests according the protocol (it may be HTTP, gRPC or mysql) used by the requests;
+- Parse SQL in the request by the parser;
+- With the parsed sql and the catalog/schema module, [DataFusion](https://github.com/apache/arrow-datafusion) can generate the logical plan;
+- With the logical plan, the corresponding `Interpreter` (`Interpreter` module  encapsulates the SQL `CRUD` operations) is created and logical plan will be executed by it;
+- For the logical plan of normal `Select` SQL, it will be executed through `SelectInterpreter`;
+- In the `SelectInterpreter` the specific query logic is executed by the `Query Engine`:
+    - Optimize the logical plan;
+    - Generate the physical plan;
+    - Optimize the physical plan;
+    - Execute the physical plan;
+- The execution of physical plan involves `Analytic Engine`:
+    - Data is obtained by `read` method of `Table` instance provided by `Analytic Engine`;
+    - The source of the table data is `SST` and `Memtable`, and the data supports being filtered by the pushed down predicates;
```

Comment:
>  and the data supports being filtered by the pushed down predicates;

and those components may support predicates to filter unnecessary data.

Reply:
> support predicates to filter ...

It sounds not very good.

Reply:
How about "the data can be filtered by the pushed down predicates."

Reply:
Sounds good.